### PR TITLE
Add benchmark report generated in summary

### PR DESCRIPTION
**Pull request by zhulinJulia24**
1. add benchmark report generated in summary.  test report is https://github.com/zhulinJulia24/lmdeploy/actions/runs/8628646975
image

2. add -x  in pr_test, when one testcase run fail pytest will fail.
3. change model path into local path, because nfs is not stable

### CODE REVIEW

File: .github/scripts/get_benchmark_summary.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import glob
+import os
+
+import fire
+import pandas as pd
+
+
+def generate_report(report_path: str):
```
Comment (RunningLeon) [inline .github/scripts/get_benchmark_summary.py:8]: could just merge the csvfiles and use add_summary to gen summary
https://github.com/InternLM/lmdeploy/blob/4b2c946776a37cce8f9936b639b9999156a9a101/.github/scripts/action_tools.py#L60

Comment (RunningLeon) [reply .github/scripts/get_benchmark_summary.py:8]: better put the function in https://github.com/InternLM/lmdeploy/blob/main/.github/scripts/action_tools.py

**Comment by zhulinJulia24** [comment]
https://github.com/zhulinJulia24/lmdeploy/actions/runs/8643455369

image

**Comment by RunningLeon** [comment]
@zhulinJulia24 hi, there are too many digits after the zero point for RPS/RPM columns, can you make it nicer with three digits?

**Comment by zhulinJulia24** [comment]
> @zhulinJulia24 hi, there are too many digits after the zero point for RPS/RPM columns, can you make it nicer with three digits?

fixed

**Review by RunningLeon — APPROVED** [review]
LGTM

File: .github/workflows/benchmark.yml [hunk 2]
```diff
@@ -162,7 +161,7 @@ jobs:
           mkdir ${result_dir}
           export CUDA_VISIBLE_DEVICES=6,7
           python3 benchmark/profile_generation.py $MODEL_PATH $MAX_ENTRY_COUNT $MODEL_FORMAT $TP_INFO -c 1 --csv ${result_dir}/generation_batch1.csv > ${result_dir}/generation_batch1.log
-          python3 benchmark/profile_generation.py $MODEL_PATH $MAX_ENTRY_COUNT $MODEL_FORMAT $TP_INFO -c 16 -ct 128 -pt 2048 --csv ${result_dir}/generation_batch16.csv > ${result_dir}/generation_batch16.log
+          python3 benchmark/profile_generation.py $MODEL_PATH $MAX_ENTRY_COUNT $MODEL_FORMAT $TP_INFO -c 16 -ct 2048 -pt 2048 --csv ${result_dir}/generation_batch16.csv > ${result_dir}/generation_batch16.log
```
Comment (lvhan028) [inline .github/workflows/benchmark.yml:164]: 只在A100上测试，没有必要测那么多concurrency settings。
这里的 MODEL_PATH 是什么？

Comment (zhulinJulia24) [reply .github/workflows/benchmark.yml]: model_path是 MODEL_PATH: /nvme/qa_test_models/${{matrix.model}}

其中matrix.model是用的github的matrix
默认覆盖的model是
'internlm/internlm2-chat-20b','internlm/internlm2-chat-20b-inner-w4a16','meta-llama/Llama-2-7b-chat-hf','meta-llama/Llama-2-7b-chat-hf-inner-w4a16'

Comment (zhulinJulia24) [reply .github/workflows/benchmark.yml]: > 只在A100上测试，没有必要测那么多concurrency settings。 这里的 MODEL_PATH 是什么？

concurrency settings已优化

File: .github/workflows/benchmark.yml [hunk 3]
```diff
@@ -160,10 +160,7 @@ jobs:
           rm -rf ${result_dir}
           mkdir ${result_dir}
           export CUDA_VISIBLE_DEVICES=6,7
-          python3 benchmark/profile_generation.py $MODEL_PATH $MAX_ENTRY_COUNT $MODEL_FORMAT $TP_INFO -c 1 --csv ${result_dir}/generation_batch1.csv > ${result_dir}/generation_batch1.log
-          python3 benchmark/profile_generation.py $MODEL_PATH $MAX_ENTRY_COUNT $MODEL_FORMAT $TP_INFO -c 16 -ct 2048 -pt 2048 --csv ${result_dir}/generation_batch16.csv > ${result_dir}/generation_batch16.log
-          python3 benchmark/profile_generation.py $MODEL_PATH $MAX_ENTRY_COUNT $MODEL_FORMAT $TP_INFO -c 32 -ct 128 2048 -pt 2048 128 --csv ${result_dir}/generation_batch32.csv > ${result_dir}/generation_batch32.log
-          python3 benchmark/profile_generation.py $MODEL_PATH $MAX_ENTRY_COUNT $MODEL_FORMAT $TP_INFO -c 64 -ct 128 -pt 128 --csv ${result_dir}/generation_batch64.csv > ${result_dir}/generation_batch64.log
+          python3 benchmark/profile_generation.py $MODEL_PATH $MAX_ENTRY_COUNT $MODEL_FORMAT $TP_INFO -c 128 128 2048 -ct 128 2048 128 -pt 128 --csv ${result_dir}/generation_batch64.csv > ${result_dir}/generation_batch64.log
```
Comment (lvhan028) [inline .github/workflows/benchmark.yml]: 不是的。是-c 128 (表示concurrency)
-ct 128 2048 128 -pt 128 2048 128
文件名：generation_batch128.csv
$MODEL_PATH 都有哪些呢？如果测试了多个模型，那么 --csv 要有所区别了

**Comment by zhulinJulia24** [comment]
嗯 刚才写错了 发现了 已经改了



---- Replied Message ----
| From | Lyu ***@***.***> |
| Date | 04/15/2024 12:03 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [InternLM/lmdeploy] Add benchmark report generated in summary (PR #1419) |

@lvhan028 commented on this pull request.

In .github/workflows/benchmark.yml:

> @@ -160,10 +160,7 @@ jobs:
           rm -rf ${result_dir}
           mkdir ${result_dir}
           export CUDA_VISIBLE_DEVICES=6,7
-          python3 benchmark/profile_generation.py $MODEL_PATH $MAX_ENTRY_COUNT $MODEL_FORMAT $TP_INFO -c 1 --csv ${result_dir}/generation_batch1.csv > ${result_dir}/generation_batch1.log
-          python3 benchmark/profile_generation.py $MODEL_PATH $MAX_ENTRY_COUNT $MODEL_FORMAT $TP_INFO -c 16 -ct 2048 -pt 2048 --csv ${result_dir}/generation_batch16.csv > ${result_dir}/generation_batch16.log
-          python3 benchmark/profile_generation.py $MODEL_PATH $MAX_ENTRY_COUNT $MODEL_FORMAT $TP_INFO -c 32 -ct 128 2048 -pt 2048 128 --csv ${result_dir}/generation_batch32.csv > ${result_dir}/generation_batch32.log
-          python3 benchmark/profile_generation.py $MODEL_PATH $MAX_ENTRY_COUNT $MODEL_FORMAT $TP_INFO -c 64 -ct 128 -pt 128 --csv ${result_dir}/generation_batch64.csv > ${result_dir}/generation_batch64.log
+          python3 benchmark/profile_generation.py $MODEL_PATH $MAX_ENTRY_COUNT $MODEL_FORMAT $TP_INFO -c 128 128 2048 -ct 128 2048 128 -pt 128 --csv ${result_dir}/generation_batch64.csv > ${result_dir}/generation_batch64.log


不是的。是-c 128 (表示concurrency)
-ct 128 2048 128 -pt 128 2048 128
文件名：generation_batch128.csv
$MODEL_PATH 都有哪些呢？如果测试了多个模型，那么 --csv 要有所区别了

File: .github/workflows/benchmark.yml (hunk 3 above)
Comment (zhulinJulia24) [reply .github/workflows/benchmark.yml]: > 不是的。是-c 128 (表示concurrency) -ct 128 2048 128 -pt 128 2048 128 文件名：generation_batch128.csv $MODEL_PATH 都有哪些呢？如果测试了多个模型，那么 --csv 要有所区别了

不用的，存储路径上有区分的

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**